rspec-core/lib/rspec/core/project_initializer/spec/spec_helper.rb

```diff
@@ -44,6 +44,11 @@
   # triggering implicit auto-inclusion in groups with matching metadata.
   config.shared_context_metadata_behavior = :apply_to_host_groups
 
+  # This setting enables Ruby deprecation warnings. It's recommended, but in some cases may
+  # be too noisy due to issues in dependencies. It can be set to `:all` or `:none`
+  # to either display all warnings or none of them.
+  config.warnings = :deprecations_only
```

Comment:
Previously, we had `config.warnings = true`, which translates to `config.warnings = :all`.
Having `:deprecations_only` here is a setback.

[Here](https://github.com/rspec/rspec/issues/37#issuecomment-2504702317) we agree that turning all by default:
> place it into our generated spec helper, with a comment explaining what it is. This has the effect of effectively being on by default
```suggestion
  config.warnings = :all
```

Reply:
@JonRowe How do you feel about defaulting to `:all` in the project initializer?

Reply:
The intent for 4 was to have the default set to `:deprecations_only` as recommended, so I moved the initialiser to suggest `:all` as an alternative.